cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -0,0 +1,86 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.support.handlers;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.api.common.HttpErrorStatusProvider;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import io.cdap.cdap.support.services.SupportBundleService;
+import io.cdap.cdap.support.status.SupportBundleConfiguration;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Support Bundle HTTP Handler.
+ */
+@Path(Constants.Gateway.API_VERSION_3)
+public class SupportBundleHttpHandler extends AbstractAppFabricHttpHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SupportBundleHttpHandler.class);
+  private final SupportBundleService bundleService;
+
+  @Inject
+  SupportBundleHttpHandler(SupportBundleService supportBundleService) {
+    this.bundleService = supportBundleService;
+  }
+
+  /**
+   * Trigger the Support Bundle Generation.
+   *
+   * @param namespace the namespace id
+   * @param application the app id
+   * @param programType the program type
+   * @param programName the program name
+   * @param run the runid of the workflow uuid of this support bundle
+   * @param maxRunsPerProgram the max num of run log for each pipeline do they prefer
+   */
+  @POST
+  @Path("/support/bundle")
+  public void createSupportBundle(HttpRequest request, HttpResponder responder,
+                                  @Nullable @QueryParam("namespace") String namespace,
+                                  @Nullable @QueryParam("application") String application,
+                                  @Nonnull @QueryParam("programType") String programType,
+                                  @Nonnull @QueryParam("programId") String programName,
+                                  @Nullable @QueryParam("run") String run,
+                                  @Nullable @QueryParam("maxRunsPerProgram") @DefaultValue("1")
+                                    Integer maxRunsPerProgram) {
+    // Establishes the support bundle configuration
+    try {
+      SupportBundleConfiguration bundleConfig =
+        new SupportBundleConfiguration(namespace, application, run, programType, programName, maxRunsPerProgram);
+      // Generates support bundle and returns with uuid
+      String uuid = bundleService.generateSupportBundle(bundleConfig);
+      responder.sendString(HttpResponseStatus.OK, uuid);
+    } catch (Exception e) {
+      LOG.error("Failed to trigger support bundle generation ", e);
+      if (e instanceof HttpErrorStatusProvider) {
```

Comment:
What if it's not? Why do you need to handle this overall? I think the common netty handlers already handle `HttpErrorStatusProvider`

Reply:
Wangyuan said it's good to return the correct error status to the front end

Reply:
Yes, but this is not Handler business. It's done by `io.cdap.cdap.common.HttpExceptionHandler` installed in `CommonNettyHttpServiceBuilder`